src/openloco/windows/toolbar_top.cpp

```diff
@@ -717,48 +717,22 @@ namespace openloco::ui::windows::toolbar_top
         }
 
         uint8_t ddIndex = 0;
-        for (uint16_t vehicleType = 0; vehicleType < vehicle_category_count; vehicleType++)
+        for (uint16_t vehicleType = 0; vehicleType < static_cast<uint8_t>(VehicleType::Count); vehicleType++)
         {
             if ((available_vehicles & (1 << vehicleType)) == 0)
                 continue;
 
-            static const uint32_t vehicle_images[] = {
-                interface_skin::image_ids::vehicle_train,
-                interface_skin::image_ids::vehicle_bus,
-                interface_skin::image_ids::vehicle_truck,
-                interface_skin::image_ids::vehicle_tram,
-                interface_skin::image_ids::vehicle_aircraft,
-                interface_skin::image_ids::vehicle_ship,
-            };
-
-            uint32_t vehicle_image = gfx::recolour(vehicle_images[vehicleType], company_colour);
-
-            static const string_id num_singular[] = {
-                string_ids::num_trains_singular,
-                string_ids::num_buses_singular,
-                string_ids::num_trucks_singular,
-                string_ids::num_trams_singular,
-                string_ids::num_aircrafts_singular,
-                string_ids::num_ships_singular,
-            };
-
-            static const string_id num_plural[] = {
-                string_ids::num_trains_plural,
-                string_ids::num_buses_plural,
-                string_ids::num_trucks_plural,
-                string_ids::num_trams_plural,
-                string_ids::num_aircraft_plural,
-                string_ids::num_ships_plural,
-            };
+            auto& interface_param = VehicleTypeInterfaceParameters.at(static_cast<VehicleType>(vehicleType));
 
+            uint32_t vehicle_image = gfx::recolour(interface_param.image, company_colour);
```

Comment:
You've replaced the vehicle image id with the build vehicle image id.

Reply:
Doh